### PR TITLE
FIX: sam parser empty case

### PR DESCRIPTION
**Pull request by mortonjt**
Not sure the best way to handle this.  Right now it returns a nested empty list. 

Specific suggestions are welcome.  I couldn't figure out how to return an empty iterator.

### CODE REVIEW

**Comment by RNAer** [comment]
tests failed

File: micronota/bfillings/diamond.py [hunk 1]
```diff
@@ -355,6 +355,8 @@ def parse_sam(diamond_res):
                    'qstart', 'sstart', 'evalue', 'bitscore', 'sseq']
         df = pd.DataFrame(columns=columns)
         for i, seq in enumerate(seqs):
+            if seq == []:
```
Comment (RNAer) [inline micronota/bfillings/diamond.py]: this is wrong.  `seq` is a `Sequence` object, should nerver been a `[]`! You should change this function.

File: micronota/parsers/sam.py [hunk 2]
```diff
@@ -219,4 +222,7 @@ def _parse_records(fh, constructor=None, **kwargs):
             req = dict(zip(_ALIGNMENT_HEADERS, req))
 
             md = merge_dicts(metadata, req, *opt)
-            yield seq, md
+            res = seq, md
+            yield res
+            res = []
+    yield res
```
Comment (RNAer) [inline micronota/parsers/sam.py]: To get a generator of empty iterator, you should use this:
```
def empty_generator():
    return
    yield
```

Comment (mortonjt) [inline micronota/parsers/sam.py]: That is awesome!!

File: micronota/bfillings/diamond.py (hunk 1 above)
Comment (mortonjt) [inline micronota/bfillings/diamond.py]: Removed.  Thank goodness

**Pull request merged**